src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts

```diff
@@ -43,14 +43,40 @@ export class ConsultationPublicResponsesComponent implements OnInit, AfterViewCh
         this.profileData = data;
         this.responseRounds = this.profileData.responseRounds;
         this.activeRoundNumber = this.getActiveRound(this.responseRounds);
-        this.responseList = data.sharedResponses.edges;
+        this.responseList = this.filterResponseData(data);
         this.publicResponsesLength = this.responseList.filter((response) => response.node.roundNumber === this.activeRoundNumber).length;
         this.roundNumberExist = this.responseList.filter((response) => response.node.roundNumber).length;
         this.checkForFragments = true;
       }
     });
   }
 
+
+  filterResponseData(data) {
```

Comment:
@preethijayanthi I think there should be another parameter here for roundNumber. In this function you're calling the getQuestions function which returns the questions from the activeRound. However if we're viewing the responses for a different round number other than the activeRound, this might cause a bug. I think both  filterResponseData and getQuestions should have a roundNumber parameter so that the responses of a particular round number will get the corresponding questions for that round

Reply:
To help you understand the previous point, each round can have different questions and hence different answers. If a Consultation has more than one round, the activeRound will be the latest round only.